### PR TITLE
Add disconnected cluster best practices guide

### PR DESCRIPTION
**Pull request by moio**
## Description

This came out of a collaboration with SUSE Support (@gboriso, @axeal) to give customer guidance when it comes to disconnected clusters.

### CODE REVIEW

File: docs/reference-guides/best-practices/rancher-managed-clusters/disconnected-clusters.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+### Best Practices for Managing Disconnected Clusters
+
+- **Cluster Availability During Rancher Upgrades**: It is recommended to have all, or at least most, managed clusters online during a Rancher upgrade. The reason is that upgrading Rancher automatically upgrades the Rancher agent software running on managed clusters. Keeping the agent and Rancher versions aligned ensures consistent functionality. Any clusters that are disconnected during the upgrade will have their agents updated as soon as they reconnect.
+
+- **Cleaning Up Disconnected Clusters**: Regularly remove clusters that will no longer reconnect to Rancher (e.g., clusters that have been decommissioned or destroyed). Keeping such clusters in the Rancher management system consumes unnecessary resources, which could impact Rancher's performance over time.
+
+- **Certificate Rotation Considerations**: When designing processes that involve regularly shutting down clusters, whether connected to Rancher or not, keep into account for certificate rotation policies. For example, RKE/RKE2/k3s clusters may rotate certificates on startup if they exceeded their lifetime.
```
Comment (btat) [inline docs/reference-guides/best-practices/rancher-managed-clusters/disconnected-clusters.md:19]:
```suggestion
- **Certificate Rotation Considerations**: When designing processes that involve regularly shutting down clusters, whether connected to Rancher or not, take into account certificate rotation policies. For example, RKE/RKE2/K3s clusters may rotate certificates on startup if they exceeded their lifetime.
```

File: versioned_docs/version-2.9/reference-guides/best-practices/rancher-managed-clusters/disconnected-clusters.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Best Practices for Managing Disconnected Clusters
+
+- **Cluster Availability During Rancher Upgrades**: It is recommended to have all, or at least most, managed clusters online during a Rancher upgrade. The reason is that upgrading Rancher automatically upgrades the Rancher agent software running on managed clusters. Keeping the agent and Rancher versions aligned ensures consistent functionality. Any clusters that are disconnected during the upgrade will have their agents updated as soon as they reconnect.
+
+- **Cleaning Up Disconnected Clusters**: Regularly remove clusters that will no longer reconnect to Rancher (e.g., clusters that have been decommissioned or destroyed). Keeping such clusters in the Rancher management system consumes unnecessary resources, which could impact Rancher's performance over time.
+
+- **Certificate Rotation Considerations**: When designing processes that involve regularly shutting down clusters, whether connected to Rancher or not, keep into account for certificate rotation policies. For example, RKE/RKE2/k3s clusters may rotate certificates on startup if they exceeded their lifetime.
```
Comment (btat) [inline versioned_docs/version-2.9/reference-guides/best-practices/rancher-managed-clusters/disconnected-clusters.md:19]:
```suggestion
- **Certificate Rotation Considerations**: When designing processes that involve regularly shutting down clusters, whether connected to Rancher or not, take into account certificate rotation policies. For example, RKE/RKE2/K3s clusters may rotate certificates on startup if they exceeded their lifetime.
```

**Review by btat — CHANGES_REQUESTED** [review]
The new disconnected-clusters.md file needs a sidebar entry:
- /docs: https://github.com/rancher/rancher-docs/blob/main/sidebars.js#L844-L848
- /version-2.9: https://github.com/rancher/rancher-docs/blob/main/versioned_sidebars/version-2.9-sidebars.json#L809-L812

**Comment by moio** [comment]
> The new disconnected-clusters.md file needs a sidebar entry:
> 
>     * /docs: [`main`/sidebars.js#L844-L848](https://github.com/rancher/rancher-docs/blob/main/sidebars.js?rgh-link-date=2024-11-05T23%3A12%3A03Z#L844-L848)
> 
>     * /version-2.9: [`main`/versioned_sidebars/version-2.9-sidebars.json#L809-L812](https://github.com/rancher/rancher-docs/blob/main/versioned_sidebars/version-2.9-sidebars.json?rgh-link-date=2024-11-05T23%3A12%3A03Z#L809-L812)

Done

**Review by btat — APPROVED** [review]

**Pull request merged**